content/scripts/build.js

```diff
@@ -310,133 +340,232 @@ class Builder {
     }
   }
 
-  start({ specificFolders = null } = {}) {
-    // This prepares this.selfHash so that when we build files, we can
-    // write down which "self hash" was used at the time.
-    if (specificFolders) {
-      // Check that they all exist and are folders
-      const allProcessed = [];
-      specificFolders.forEach((folder) => {
-        if (!fs.existsSync(folder)) {
-          throw new Error(`${folder} does not exist`);
-        }
-        if (!fs.statSync(folder).isDirectory()) {
-          throw new Error(`${folder} is not a directory`);
+  getSource(folder) {
+    if (!fs.existsSync(folder)) {
+      throw new Error(`${folder} does not exist`);
+    }
+    if (!fs.statSync(folder).isDirectory()) {
+      throw new Error(`${folder} is not a directory`);
+    }
+    const source = this.sources.entries().find((source) => {
+      return folder.startsWith(source.filepath);
+    });
+    if (!source) {
+      throw new Error(`Unable to find the source based on ${folder}`);
+    }
+    return source;
+  }
+
+  reportMacroRenderingFlaws() {
```

Comment:
This one is going to need a refactor later. As we discussed. We should stuff the flaws into the `doc` (which we use when dumping the `index.json`).